src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,226 @@
+package frc.robot.subsystems.intake;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.lib.team3061.leds.LEDs;
+import frc.lib.team3061.leds.LEDsRIO;
+import frc.lib.team3061.leds.LEDs.IntakeLEDState;
+
+import org.littletonrobotics.junction.Logger;
+
+public class Intake extends SubsystemBase {
+
+  private final IntakeIO io;
+  private final IntakeIOInputsAutoLogged inputs = new IntakeIOInputsAutoLogged();
+  private final LEDs leds;
+
+  // Intakes, Drum, Kicker
+  enum IntakeState {
+    EMPTY, // 0, 0, 0
+    IN_BETWEEN_DRUM_AND_KICKER,
+    IN_BETWEEN_INTAKE_AND_DRUM,
+    NOTE_IN_KICKER, // 0, 0, 1
+    NOTE_IN_DRUM, // 0, 1, 0
+    NOTE_IN_INTAKE, // 1, 0, 0
+    NOTE_IN_INTAKE_AND_DRUM, // 1, 1, 0
+    EMERGENCY, // 0, 1, 1 or 1, 0, 1 or 1, 1, 1
+  }
+
+  private IntakeState intakeState;
+  // will update intake state in different areas / methods based on the sensors,
+  // but first just running through the logic of what we would do in these cases
+
+  private boolean manualOverrideEnabled;
+
+  public Intake(IntakeIO io) {
+    this.io = io;
+    leds = LEDs.getInstance();
+    intakeState = IntakeState.EMPTY;    
+
+    leds.setIntakeLEDState(IntakeLEDState.WAITING_FOR_GAME_PIECE);
+    this.intakeGamePiece();
```

Comment:
register a system test method; you can implement it later. Let's discuss in person.

```
FaultReporter.getInstance().registerSystemCheck(INTAKE, getSystemCheckCommand());
```